core/block.js

```diff
@@ -695,9 +695,9 @@ Blockly.Block.prototype.getVarModels = function() {
   for (var i = 0, input; input = this.inputList[i]; i++) {
     for (var j = 0, field; field = input.fieldRow[j]; j++) {
       if (field instanceof Blockly.FieldVariable) {
-        // TODO (#1199): When we switch to tracking variables by ID,
-        // update this.
-        var model = this.workspace.getVariable(field.getValue(), '');
+        var model = this.workspace.getVariableById(field.getValue());
+        // Check if the variable actually exists (and isn't just a potential
+        // variable).
```

Comment:
What is the intuition behind 'potential variables' vs. actual variables? 

Is it that we don't want to preclude someone (for example) from creating a variable 'i' just because the flyout may contain a for loop that initializes a local variable 'i'?